src/asm/symbol.c

```diff
@@ -288,15 +295,47 @@ void sym_Purge(char const *symName)
 	} else if (isReferenced(sym)) {
 		error("Symbol \"%s\" is referenced and thus cannot be purged\n", symName);
 	} else {
-		// Do not keep a reference to the label's name after purging it
-		if (sym->name == labelScope)
-			sym_SetCurrentSymbolScope(NULL);
-
-		// FIXME: this leaks sym->macro for SYM_EQUS and SYM_MACRO, but this can't
-		// free(sym->macro) because the expansion may be purging itself.
-		hash_RemoveElement(symbols, sym->name);
-		// TODO: ideally, also unref the file stack nodes
-		free(sym);
+		struct SymbolListNode *end = purgeList;
+
+		while (end && end->next)
+			end = end->next;
```

Comment:
:wrench: Seeking to the end of the purge list each time could be avoided if you prepend each symbol to the beginning, instead of appending to the end.